File: ruby/simple_calculator/simple_calculator_6.rb
# This is a simple calculator program written in Ruby.
# The purpose of this program is to perform basic arithmetic operations
# such as addition, subtraction, multiplication, and division.

def add(a, b)
    a + b
end

def subtract(a, b)
    a - b
end

def multiply(a, b)
    a * b
end

def divide(a, b)
    a / b
end

puts "Welcome to the Simple Calculator"
puts "Please enter the first number:"
num1 = gets.chomp.to_f

puts "Please enter the second number:"
num2 = gets.chomp.to_f

puts "Please choose an operation: add, subtract, multiply, divide"
operation = gets.chomp

result = case operation
                 when "add"
                     add(num1, num2)
                 when "subtract"
                     subtract(num1, num2)
                 when "multiply"
                     multiply(num1, num2)
                 when "divide"
                     divide(num1, num2)
                 else
                     "Invalid operation"
                 end

puts "The result is: #{result}"

File: javascript/simple_calculator/simple_calculator_6.js
/**
 * This is a simple calculator program written in JavaScript.
 * It performs basic arithmetic operations such as addition, subtraction,
 * multiplication, and division based on user input.
 */

function simpleCalculator() {
    const operation = prompt("Enter operation (+, -, *, /): ");
    const num1 = parseFloat(prompt("Enter first number: "));
    const num2 = parseFloat(prompt("Enter second number: "));

    let result;

    switch (operation) {
        case '+':
            result = num1 + num2;
            break;
        case '-':
            result = num1 - num2;
            break;
        case '*':
            result = num1 * num2;
            break;
        case '/':
            result = num1 / num2;
            break;
        default:
            alert("Invalid operation");
            return;
    }

    alert("The result is: " + result);
}

simpleCalculator();

